pkg/kubelet/pod_workers.go

```diff
@@ -981,25 +981,29 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	// enforce the restriction that a grace period can only decrease and track whatever our value is,
 	// then ensure a calculated value is passed down to lower levels
 	gracePeriod := status.gracePeriod
-	// this value is bedrock truth - the apiserver owns telling us this value calculated by apiserver
-	if override := pod.DeletionGracePeriodSeconds; override != nil {
-		if gracePeriod == 0 || *override < gracePeriod {
+
+	// make a best effort to default this value to the pod's desired intent, in the event
+	// the kubelet provided no requested value (graceful termination?)
+	if override := pod.Spec.TerminationGracePeriodSeconds; override != nil {
+		if *override < gracePeriod {
 			gracePeriod = *override
 		}
 	}
+
+	// this value is bedrock truth - the apiserver owns telling us this value calculated by apiserver
```

Comment:
I think this also needs to be clarified better. The current comments I think are very confusing. we have a "bedrock truth" here and we had a "make a best effort to default to this value" in the other check.

Reply:
The term "bedrock truth" was difficult to understand. Thank you for the fix. I think the new comment makes better sense.

For context, it was better to change how this is stated because:
- it's not a widely used term in general, I think.
- "bedrock truth" as a term is only used in this part of the code. Not used anywhere else.

Reply:
Thank you for your suggestion. My idea is(please see 1st line):
```
	// When grace period is specified by adding --grace-period or --force option, this value is stored in pod.DeletionGracePeriodSeconds
	// the apiserver owns telling us this value calculated by apiserver
	if override := pod.DeletionGracePeriodSeconds; override != nil {
		gracePeriod = *override
	}
```
What do you think?